source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.cc

```diff
@@ -209,6 +217,27 @@ void GradientController::cancelLatencySample() {
   --num_rq_outstanding_;
 }
 
+void GradientController::updateConcurrencyLimit(const uint32_t new_limit) {
+  const auto old_limit = concurrency_limit_.load();
+  concurrency_limit_.store(new_limit);
+  stats_.concurrency_limit_.set(concurrency_limit_.load());
+
+  if (!inMinRTTSamplingWindow() && old_limit == config_.minConcurrency() &&
+      new_limit == config_.minConcurrency()) {
+    ++consecutive_min_concurrency_set_;
+  } else {
+    consecutive_min_concurrency_set_ = 0;
+  }
+
+  // If the concurrency limit is being set to the minimum value for the 5th consecutive sample
+  // window while not in the middle of a minRTT measurement, this might be indicative of an
+  // inaccurate minRTT measurement. Since the limit is already where it needs to be for a minRTT
+  // measurement, we should measure it again.
+  if (consecutive_min_concurrency_set_ >= 5) {
+    min_rtt_calc_timer_->enableTimer(std::chrono::milliseconds(0));
```

Comment:
This won't end up spin looping because if this happens while not already in a window, this will start a process that waits for requests/responses to happen, right? Can you maybe put a clarifying comment about why this won't spin loop? This is also why you need the various `inMinRTTSamplingWindow` checks, right?

Reply:
I think any issues around spin looping are prevented first thing in `enterMinRTTSamplingWindow`. Added a comment to clarify this, but let me know if I addressed the right concern.